### PR TITLE
[distributions] Refactor `_log_sum_exp`

### PR DESCRIPTION
**Pull request by alicanb**
This PR removes `distributions.utils._log_sum_exp` in favor of `torch.logsumexp`. Also fixes some warnings with `reduce` arg. in `binary_cross_entropy_with_logits`

### CODE REVIEW

File: torch/distributions/bernoulli.py [hunk 1]
```diff
@@ -77,10 +77,10 @@ def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
         logits, value = broadcast_all(self.logits, value)
-        return -binary_cross_entropy_with_logits(logits, value, reduce=False)
+        return -binary_cross_entropy_with_logits(logits, value, reduction='none')
```
Comment (vishwakftw) [inline torch/distributions/bernoulli.py]: These are taken care of in #9162 .

File: torch/distributions/bernoulli.py [hunk 2]
```diff
@@ -77,10 +77,10 @@ def log_prob(self, value):
...
 
     def entropy(self):
-        return binary_cross_entropy_with_logits(self.logits, self.probs, reduce=False)
+        return binary_cross_entropy_with_logits(self.logits, self.probs, reduction='none')
```
Comment (vishwakftw) [inline torch/distributions/bernoulli.py]: Same as above.

File: torch/distributions/geometric.py [hunk 3]
```diff
@@ -74,4 +74,4 @@ def log_prob(self, value):
         return value * (-probs).log1p() + self.probs.log()
 
     def entropy(self):
-        return binary_cross_entropy_with_logits(self.logits, self.probs, reduce=False) / self.probs
+        return binary_cross_entropy_with_logits(self.logits, self.probs, reduction='none') / self.probs
```
Comment (vishwakftw) [inline torch/distributions/geometric.py]: Same as above.

File: torch/distributions/bernoulli.py (hunk 1 above)
Comment (alicanb) [reply torch/distributions/bernoulli.py]: Thanks @vishwakftw , I'm gonna revert the last commit once #9162 is merged.

**Review by apaszke — APPROVED** [review]

**Pull request closed without merging**